src/main/java/net/snowflake/client/jdbc/SnowflakeColumn.java

```diff
@@ -0,0 +1,64 @@
+package net.snowflake.client.jdbc;
+
+import static java.lang.annotation.ElementType.FIELD;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface SnowflakeColumn {
+
+  /**
+   * (Optional) The name for a column in database,
+   *
+   * <p>The default value is empty string. Provided name can override SqlData field name
+   */
+  String name() default "";
+
+  /**
+   * (Optional) The snowflake type for a column
+   *
+   * <p>The default value is empty string Provided type can override default type
+   */
+  String type() default "";
+
+  /**
+   * (Optional) The snowflake nullable flag for a column
+   *
+   * <p>The default value is true Provided value can override default nullable value
+   */
+  boolean nullable() default true;
+
+  /**
+   * (Optional) The length for a column of SQL type {@code varchar} or {@code binary}, or of similar
+   * database-native type.
+   *
+   * <p>Applies only to columns of exact varchar and binary type.
+   *
+   * <p>The default value {@code -1} indicates that a provider-determined length should be inferred.
+   */
+  int length() default -1;
```

Comment:
Should we have byte length as well?

Reply:
I'm not sure. I thought length is for text columns whereas byte length is for some binary columns?

Reply:
I added byteLength field